src/services/refactors/moveToFile.ts

```diff
@@ -430,6 +430,10 @@ export function updateImportsInOtherFiles(
                 deleteUnusedImports(sourceFile, importNode, changes, shouldMove); // These will be changed to imports from the new file
 
                 const pathToTargetFileWithExtension = resolvePath(getDirectoryPath(oldFile.path), targetFileName);
+
+                // no self-imports
+                if (sourceFile.fileName === pathToTargetFileWithExtension) return;
+

```

Comment:
```suggestion
                const pathToTargetFileWithExtension = resolvePath(getDirectoryPath(getNormalizedAbsolutePath(oldFile.fileName, program.getCurrentDirectory())), targetFileName);
                // no self imports
                if (getStringComparer(!program.useCaseSensitiveFileNames())(pathToTargetFileWithExtension, sourceFile.fileName) === Comparison.EqualTo) return; 

```

Reply:
@navya9singh  we also need to replace usage of `sourceFile.path` in `makeImportOrReqiure` so its not using canonical path but actual file name: 
```ts
 const pathToTargetFile = resolvePath(getDirectoryPath(sourceFile.path), targetFileNameWithExtension);
```
to 
```ts
const pathToTargetFile = resolvePath(getDirectoryPath(getNormalizedAbsolutePath(sourceFile.fileName, program.getCurrentDirectory())), targetFileNameWithExtension);

```

Reply:
made the changes